src/HaroohiePals.Gui/ImGuiController.cs

```diff
@@ -160,390 +124,441 @@ void main()
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Undo_16x, FontAwesome6.ArrowRotateLeft[0], 16));
+        _icons.Add(new ImGuiIconGlyph(Resources.Icons.Redo_16x, FontAwesome6.ArrowRotateRight[0], 16));
+
+        if (icons != null)
+            _icons.AddRange(icons);
+    }
 
-            io.Fonts.ClearTexData();
-        }
+    private void InitFonts()
+    {
+        bool isDefaultFont = true;
 
-        /// <summary>
-        /// Renders the ImGui draw list data.
-        /// </summary>
-        public void Render()
+        foreach (var font in _fonts)
         {
-            if (_frameBegun)
+            AddFont(font, GetImFontConfig());
+
+            // Merge icon font only for the default font (first font)
+            if (isDefaultFont)
             {
-                _frameBegun = false;
-                ImGui.Render();
-                RenderImDrawData(ImGui.GetDrawData());
+                var iconRanges = new ushort[] { FontAwesome6.IconMin, FontAwesome6.IconMax, 0 };
+                AddFont(_iconFont, GetImFontConfig(true), iconRanges);
             }
+
+            isDefaultFont = false;
         }
+    }
 
-        /// <summary>
-        /// Updates ImGui input and IO configuration state.
-        /// </summary>
-        public void Update(GameWindow wnd, float deltaSeconds)
+    private ImFontConfigPtr GetImFontConfig(bool isIconFont = false)
+    {
+        ImFontConfigPtr conf;
+        unsafe
         {
-            if (_frameBegun)
-            {
-                ImGui.Render();
-            }
+            conf = new ImFontConfigPtr(ImGuiNative.ImFontConfig_ImFontConfig());
+        }
 
-            SetPerFrameImGuiData(deltaSeconds);
-            UpdateImGuiInput(wnd);
+        conf.PixelSnapH = true;
+        conf.OversampleH = 1;
+        conf.OversampleV = 1;
 
-            _frameBegun = true;
-            ImGui.NewFrame();
-            //ImGuizmo.BeginFrame();
+        if (isIconFont)
+        {
+            conf.MergeMode = true;
+            conf.GlyphOffset.Y = 1;
+            conf.GlyphMinAdvanceX = 16f; // Use if you want to make the icon monospaced
         }
 
-        /// <summary>
-        /// Sets per-frame data based on the associated window.
-        /// This is called by Update(float).
-        /// </summary>
-        private void SetPerFrameImGuiData(float deltaSeconds)
+        return conf;
+    }
+
+    private unsafe void AddFont(ImGuiFont font, in ImFontConfigPtr fontConfig, ushort[] iconRanges = null)
+    {
+        var io = ImGui.GetIO();
+
+        fixed (byte* p = font.Data)
         {
-            ImGuiIOPtr io = ImGui.GetIO();
-            io.DisplaySize = new System.Numerics.Vector2(
-                _windowWidth / _scaleFactor.X,
-                _windowHeight / _scaleFactor.Y);
-            io.DisplayFramebufferScale = _scaleFactor;
-            io.DeltaTime = deltaSeconds; // DeltaTime is in seconds.
+            fixed (ushort* rang = iconRanges)
```

Comment:
Maybe this can just be placed below the other fixed, before the { }. Without extra identation